pytorch_lightning/loops/utilities.py

```diff
@@ -62,6 +63,18 @@ def _extract_hiddens(training_step_output: STEP_OUTPUT, truncated_bptt_steps: in
     return hiddens
 
 
+# rank_zero_warning of type UserWarning for max_epochs if not set
+def _parse_max_epochs_and_steps(max_steps, max_epochs, max_time, min_steps, min_epochs):
```

Comment:
typing ?

Reply:
My bad, I will update. So used to not using typing 😄